content/posts/2021-04-27-server-side-rendering-with-netlify/2021-04-27-server-side-rendering-with-netlify.mdx

```diff
@@ -0,0 +1,160 @@
+
+```sh
+yarn netlify dev -d public -f functions
+```
+
+Opening the server on http://localhost:8888 would just lead to a 404 error page.
+Let’s re-use the redirect rule we just learned earlier:
+
+```sh
+# file public/_redirects
+/*         /.netlify/functions/render              200
+```
+
+Refreshing the page now would actually not return a 404 page, instead, we get:
+
+```sh
+TypeError: lambdaFunc[lambdaHandler] is not a function
+```
+
+The problem is, we haven’t yet defined our Netlify function:
+
+```javascript
+// file functions/render/index.js
+exports.handler = async () => {
+  return {
+    statusCode: 200,
+    headers: {
+      'content-type': 'text/html; charset=utf-8',
+    },
+    body: `
+      <html>
+        <body>
+          <h1>Hello World :)</h1>
+          <p>${new Date().toISOString()}</p>
+        </body>
+      </html>
+    `,
+  };
+};
+```
+
+This change should return a web page with a headline and a paragraph. Pretty
+cool! You could even use `curl` to send requests and you would still see the
+current timestamp in the response. Let’s extend the feature set and add some
+styling:
+
+```javascript
+// file functions/render/index.js
+exports.handler = async () => {
+  return {
+    statusCode: 200,
+    headers: {
+      'content-type': 'text/html; charset=utf-8',
+    },
+    body: `
+      <html>
+        <head>
+          <link rel="stylesheet" href="styles.css" />
+        </head>
+        <body>
+          <h1>Hello World :)</h1>
+          <p>${new Date().toISOString()}</p>
+        </body>
+      </html>
+    `,
+  };
+};
+```
+
+And the CSS file:
+
+```css
+/*  file public/styles.css */
+body {
+  background-color: antiquewhite;
+}
+```
+
+This brings us to the next awesome feature! With the redirect rule we defined
+earlier, you might think all traffic has to go through the Netlify function, but
+that’s not true. With the configuration we did, Netlify checks if the requested
+file exists statically and only if it doesn’t, Netlify calls the function. Isn’t
+that cool? That means with just one simple redirect rule, we can build hybrid
+solutions, have a web server in place and still static assets like CSS or JS. We
+could even have a mix of static HTML files and dynamic routes.
+
+## Takeaways
+
+Should we use this technology to solve all our problems? Probably not. It’s just
+good to keep it in might. Netlify provides a super convenient solution to easily
+create a web server without VMs, EC2s, Kubernetes Pods or Docker Images. We
+should still consider static hosting, frameworks that do the heavy lifting for
+us and avoid reinventing the wheel, but it’s good to understand the techniques
+in the background. Still, for simple prototypes or quick solutions, this could
+be very interesting.
+
+BTW, I created a more complex Hacker News clone to demonstrate server-side
+rendering with old-fashioned template engines. The web server is back! 😂
+
+https://github.com/yetanother-blog/netlify-ssr-example
```

Comment:
since I build this broken link here we could try to shorten it for now with:
`[Netlify SSR example](https://github.com/yetanother-blog/netlify-ssr-example)`

Reply:
Yeah, I fixed it 😇